src/node/coinstats.cpp

```diff
@@ -82,7 +83,11 @@ static void ApplyStats(CCoinsStats& stats, const uint256& hash, const std::map<u
     stats.nTransactions++;
     for (auto it = outputs.begin(); it != outputs.end(); ++it) {
         stats.nTransactionOutputs++;
-        stats.nTotalAmount += it->second.out.nValue;
+        if (stats.total_amount.has_value() && !AdditionOverflow(*stats.total_amount, it->second.out.nValue)) {
```

Comment:
It seems cleaner to factor this out. What about an `AddOverflowDetect` in `util/overflow.h` that implements the "add numbers and collapse to nullopt on overflow" logic?

Reply:
Thx, done